### ISSUE TITLE
HA Yellow HB-RF-USB-2 / HM-MOD-RPI-PCB not found

### ISSUE THREAD

**Issue by H3xF2x** [jens-maus/RaspberryMatic#2307]
### Describe the issue you are experiencing

Hi,

I tried to migrate my Raspberrymatic Installation from a Waveshare CM4 board to Home Assistant Yellow.
Everything is up and running, except the HB-RF-USB-2 with HM-MOD-RPI-PCB connected to the Yellow.
The HB-RF-USB-2 is not found at all by Raspberrymatic.

The HB-RF-USB-2 recognized by HA OS - it will be displayed in the terminal window with "lsusb" or "dmsg grep usb"
(Sorry, I'm not able to select and copy the terminal output, but I'm shure it is there :-/ )

HA OS is 10.1 on the yellow.

### Describe the behavior you expected

Same behaviour as with to old hardware.

### Steps to reproduce the issue

1. Plug HB-RF-USB-2 / HM-MOD-RPI-PCB into USB of Yellow
2. Powering on / booting
3. HB-RF-USB-2 / HM-MOD-RPI-PCB not found

### What is the version this bug report is based on?

3.69.6.20230407

### Which base platform are you running?

ha-addon (HomeAssistant Add-on)

### Which HomeMatic/homematicIP radio module are you using?

HM-CFG-USB-2

### Anything in the logs that might be useful for us?

```txt
Log of raspberrymatic on the yellow

Mounting /data as /usr/local (Home Assistant Add-On): OK
Starting watchdog...
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Saving 256 bits of creditable seed for next boot
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.0, OK
Identifying Homematic RF-Hardware: ...HmRF: n/a, HmIP: n/a, OK
Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found
Starting hs485dLoader: OK
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting sshd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...cryptEnabled trueOK
Setting LAN Gateway keys: OK
Starting hs485d: OK
Starting multimacd: not required
Starting rfd: no BidCos-RF hardware found
Starting HMIPServer: ........OK
Starting ReGaHss: .OK
Starting CloudMatic: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.69.6.20230407 (raspmatic_oci_arm64)
```

**Comment by H3xF2x** [comment]
Still no success: I reinstalled HAOS and the Raspberry Addon.
HAOS is 10.1 64bit.

The HB-RF-USB-2 is recognised by the host:
BUSNUM: '001'
DEVNAME: /dev/bus/usb/001/003
DEVNUM: '003'
DEVPATH: /devices/platform/soc/fe980000.usb/usb1/1-1/1-1.1
DEVTYPE: usb_device
DRIVER: usb
ID_BUS: usb
ID_FOR_SEAT: usb-platform-fe980000_usb-usb-0_1_1
ID_MODEL: HB-RF-USB-2
ID_MODEL_ENC: HB-RF-USB-2
ID_MODEL_ID: 8c07
ID_PATH: platform-fe980000.usb-usb-0:1.1
ID_PATH_TAG: platform-fe980000_usb-usb-0_1_1
ID_REVISION: '0200'
ID_SERIAL: Alexander_Reinert_HB-RF-USB-2_7aaffae4fb9dec11a66c738a43a0c072
ID_SERIAL_SHORT: 7aaffae4fb9dec11a66c738a43a0c072
ID_USB_INTERFACES: ':ff0000:'
ID_USB_MODEL: HB-RF-USB-2
ID_USB_MODEL_ENC: HB-RF-USB-2
ID_USB_MODEL_ID: 8c07
ID_USB_REVISION: '0200'
ID_USB_SERIAL: Alexander_Reinert_HB-RF-USB-2_7aaffae4fb9dec11a66c738a43a0c072
ID_USB_SERIAL_SHORT: 7aaffae4fb9dec11a66c738a43a0c072
ID_USB_VENDOR: Alexander_Reinert
ID_USB_VENDOR_ENC: Alexander\x20Reinert
ID_USB_VENDOR_ID: 10c4
ID_VENDOR: Alexander_Reinert
ID_VENDOR_ENC: Alexander\x20Reinert
ID_VENDOR_ID: 10c4
MAJOR: '189'
MINOR: '2'
PRODUCT: 10c4/8c07/200
SUBSYSTEM: usb
TAGS: ':seat:'
TYPE: 0/0/0
USEC_INITIALIZED: '7921266'

Can someone please tell me why the HB-RF-USB-2 is not found by Raspberrmatic?

Thanks

**Comment by jens-maus** [comment]
Please show the output of `lsusb` on the HA host when having the `HB-RF-USB-2` connectioned. Also also please show the end of your `dmesg` kernel log messages once you connected the `HB-RF-USB-2` freshly.

**Comment by H3xF2x** [comment]
Hello,

thank for the reply.

lsusb:
`Bus 001 Device 001: ID 1d6b:0002
Bus 001 Device 003: ID 10c4:8c07
Bus 001 Device 002: ID 1a40:0101
`

dmesg:
`[  164.630493] audit: type=1334 audit(1683660460.828:304): prog-id=75 op=UNLOAD
[  292.515738] usb 1-1.1: new full-speed USB device number 3 using dwc2
[  292.737800] usb 1-1.1: New USB device found, idVendor=10c4, idProduct=8c07, bcdDevice= 2.00
[  292.737824] usb 1-1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  292.737832] usb 1-1.1: Product: HB-RF-USB-2
[  292.737839] usb 1-1.1: Manufacturer: Alexander Reinert
[  292.737846] usb 1-1.1: SerialNumber: 7aaffae4fb9dec11a66c738a43a0c072
[  292.758904] generic_raw_uart: loading out-of-tree module taints kernel.
[  292.763569] usb 1-1.1: DKey: 3465616666616137 3131636564396266 6138333763363661 3237306330613334
[  292.763588] usb 1-1.1: HKey: 0175a39e3365dc44 c830c0bceaad0794 60f49dfb4344eae0 77d4300e55d5bc7b
[  292.763596] usb 1-1.1: SKey: 8242078a1a4b2457 f0ac22b9e6d7cbef fa72d70c8924ccbf 1873db15343b46d3
[  292.763602] usb 1-1.1: SKey: 6419474dfd4fbfa5 902e51a4712db8db 9bf1ff3b05f07930 8ee1a2dcb397561b
[  292.763609] usb 1-1.1: PKey: f50048a8d7cc3f44 ae8460d13cf61d60 2dc161f5272a5876 88c996b27737fbae
[  292.763616] usb 1-1.1: PKey: dd24b0aced521e88 cc457acb4b7c49a4 6745032ab2079847 becfb9f3af25d41f
[  292.767034] usb 1-1.1: Successfully verified device signature
[  292.767057] usb 1-1.1: Found HB-RF-USB-2 with serial 7aaffae4fb9dec11a66c738a43a0c072 at usb-fe980000.usb-1.1
[  292.771239] raw-uart raw-uart: Reset radio module
[  292.887732] raw-uart raw-uart: Registered new raw-uart device using underlying device HB-RF-USB-2@usb-fe980000.usb-1.1.
[  292.890471] usbcore: registered new interface driver hb_rf_usb_2
[  294.856278] audit: type=1334 audit(1683660591.050:305): prog-id=78 op=LOAD
[  294.856622] audit: type=1334 audit(1683660591.050:306): prog-id=53 op=UNLOAD
[  294.858808] audit: type=1334 audit(1683660591.050:307): prog-id=79 op=LOAD
[  294.858903] audit: type=1300 audit(1683660591.050:307): arch=c00000b7 syscall=280 success=yes exit=12 a0=5 a1=4000134220 a2=78 a3=0 items=0 ppid=126 pid=6253 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
[  294.858963] audit: type=1327 audit(1683660591.050:307): proctitle=72756E63002D2D726F6F74002F7661722F72756E2F646F636B65722F72756E74696D652D72756E632F6D6F62792F00757064617465002D2D7265736F7572636573002D00313834303731376131313664653838616166643761633739643137323364643162633362316635393835663164383038353865396463343331643864
[  294.859230] audit: type=1334 audit(1683660591.050:308): prog-id=80 op=LOAD
[  294.859298] audit: type=1300 audit(1683660591.050:308): arch=c00000b7 syscall=280 success=yes exit=14 a0=5 a1=4000133fb0 a2=78 a3=0 items=0 ppid=126 pid=6253 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
[  294.859339] audit: type=1327 audit(1683660591.050:308): proctitle=72756E63002D2D726F6F74002F7661722F72756E2F646F636B65722F72756E74696D652D72756E632F6D6F62792F00757064617465002D2D7265736F7572636573002D00313834303731376131313664653838616166643761633739643137323364643162633362316635393835663164383038353865396463343331643864
[  294.859428] audit: type=1334 audit(1683660591.050:309): prog-id=80 op=UNLOAD
[  294.859586] audit: type=1334 audit(1683660591.050:310): prog-id=79 op=UNLOAD`

HTH

**Comment by Baxxy13** [comment]
The Output looks correct for me. Slightly different fromm an Pi4B but not wrong.

Do you use the HB-RF-USB-2/HM-MOD-RPI-PCB on your waveshare and it worked? Or was the HM-MOD-RPI-PCB directly stacked?

And just top be sure... The HM-MOD-RPI-PCB is stacked in the right direction on the HB-RF-USB-2?
You could show a picture.

**Comment by Baxxy13** [comment]
OK, good. That proves that the combi is generally working.
I testet yesterday the exact same combi on Pi4B with freshly installed HA-OS 10.1 64bit. 
No problems.
The difference i saw was the used usb-driver. Don't know if this may be relevant. 
Unfortunately i did not save my logs, maybe i check it later again.

**Comment by jens-maus** [comment]
Well, to further investigate this issue one would have to ssh login to the raspberrymatic add-on and then check `lsusb` within the container and also check if the `/dev/raw-uart` device was properly loaded/initialized and then execute `detect_radio_module` with the device node to raw-uart and see if this outputs anything....

**Comment by Baxxy13** [comment]
Can you give the exact commands for testing?

**Comment by Baxxy13** [comment]
Ok, some insights from the working Pi4B-2GB with HA-OS 10.1 64bit:

dmesg host HB-RF-USB-2 plugged in:
```
[  277.668098] usb 1-1.3: new full-speed USB device number 4 using xhci_hcd
[  277.775421] usb 1-1.3: New USB device found, idVendor=10c4, idProduct=8c07, bcdDevice= 1.01
[  277.775456] usb 1-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  277.775474] usb 1-1.3: Product: HB-RF-USB-2
[  277.775490] usb 1-1.3: Manufacturer: Alexander Reinert
[  277.775505] usb 1-1.3: SerialNumber: 5261a7033d32ea11a8da1576472be6b9
[  277.812000] generic_raw_uart: loading out-of-tree module taints kernel.
[  277.822064] usb 1-1.3: DKey: 3330376131363235 3131616532336433 3637353161643861 3962366562323734
[  277.822088] usb 1-1.3: HKey: 020d51f4cbcc3aeb 6a18dc20dcb6c31b ff716ad560a376b5 013d83556a988c5b
[  277.822098] usb 1-1.3: SKey: 0000000000000000 0000000000000000 0000000000000000 0000000000000000
[  277.822108] usb 1-1.3: SKey: 0000000000000000 0000000000000000 0000000000000000 0000000000000000
[  277.822118] usb 1-1.3: PKey: f50048a8d7cc3f44 ae8460d13cf61d60 2dc161f5272a5876 88c996b27737fbae
[  277.822127] usb 1-1.3: PKey: dd24b0aced521e88 cc457acb4b7c49a4 6745032ab2079847 becfb9f3af25d41f
[  277.822140] usb 1-1.3: Could not verify device signature
[  277.822212] usb 1-1.3: Found HB-RF-USB-2 with serial 5261a7033d32ea11a8da1576472be6b9 at usb-0000:01:00.0-1.3
[  277.824677] raw-uart raw-uart: Reset radio module
[  277.944178] raw-uart raw-uart: Registered new raw-uart device using underlying device HB-RF-USB-2@usb-0000:01:00.0-1.3.
[  277.949348] usbcore: registered new interface driver hb_rf_usb_2
```
ssh into running RM-HA-AddOn:
```
root@de838cd8-raspberrymatic-dev:~# lsusb
Bus 001 Device 001: ID 1d6b:0002
Bus 001 Device 004: ID 10c4:8c07
Bus 001 Device 002: ID 2109:3431
Bus 002 Device 002: ID 0781:5567
Bus 002 Device 001: ID 1d6b:0003
Bus 001 Device 003: ID 046d:c31c

root@de838cd8-raspberrymatic-dev:~# monit stop rfd
root@de838cd8-raspberrymatic-dev:~# monit stop multimacd
root@de838cd8-raspberrymatic-dev:~# detect_radio_module --debug /dev/raw-uart
14:42:11.388780 Raw UART device: HB-RF-USB-2@usb-0000:01:00.0-1.3
14:42:11.508054 Sucessfully resetted radio module.
14:42:11.508438 Sending HM frame:  fd 00 03 fe 00 01 14 1e
14:42:11.508560 Received HM frame: fd 00 0c 00 00 00 43 6f 5f 43 50 55 5f 42 4c 72 51
14:42:11.508626 Sending HM frame:  fd 00 03 fe 01 01 92 1d
14:42:14.508753 Sending HM frame:  fd 00 03 00 02 00 94 03
14:42:14.513354 Received HM frame: fd 00 0d 00 02 04 02 43 6f 5f 43 50 55 5f 42 4c bf f9
14:42:14.513404 Sending HM frame:  fd 00 03 00 03 03 12 0a
14:42:15.045724 Received HM frame: fd 00 04 00 03 04 01 99 01
14:42:15.084045 Received HM frame: fd 00 10 fe 01 00 44 75 61 6c 43 6f 50 72 6f 5f 41 70 70 b7 36
14:42:15.084142 Sending HM frame:  fd 00 03 01 04 09 00 22
14:42:15.187952 Received HM frame: fd 00 05 01 04 04 01 03 96 27
14:42:15.188060 Sending HM frame:  fd 00 03 01 05 02 06 18
14:42:15.293411 Received HM frame: fd 00 0d 01 05 04 01 02 08 06 01 00 03 01 14 03 b2 6a
14:42:15.293504 Sending HM frame:  fd 00 03 02 06 01 0c 2e
14:42:15.298914 Received HM frame: fd 00 07 02 06 06 01 b5 58 9c fd e1
14:42:15.298970 Sending HM frame:  fd 00 03 fe 07 04 86 03
14:42:15.305511 Received HM frame: fd 00 10 fe 07 05 01 30 14 f7 11 a0 61 a7 d7 09 8e 21 fb 03 99
14:42:15.305588 Sending HM frame:  fd 00 03 03 08 08 28 0c
14:42:15.310626 Received HM frame: fd 00 07 03 08 01 01 58 48 b5 40 07
14:42:15.310681 Sending HM frame:  fd 00 03 03 09 07 ae 2d
14:42:15.317019 Received HM frame: fd 00 0e 03 09 01 01 4f 45 51 30 33 30 39 37 35 35 6f b9
14:42:15.436055 Sucessfully resetted radio module.
HM-MOD-RPI-PCB OEQ0309755 3014F711A061A7D7098E21FB 0x5848B5 0xB5589C 2.8.6
```

**Comment by reddevil156** [comment]
Hello everyone,

as I wrote in another thread I'm trying to get the same combination to work on a proxmox server without success.
The pcb and the module are working fine when connected to a raspberry pi 3 but nowhere else. I used several versions of proxmox and RaspberryMatic and even debmatic didn't work.

On the HB-RF-USB-2 is a uart converter chip (2102N) which provides 3.3V to the HM-MOD-RPI-PCB if it is enumerated and configured correctly and this doesn't happen here.

I changed the buildroot of RaspberryMatic and added the kernel option "Dynamic Debug" for debugging purposes. Attached is a dmesg log with usbcore, xhci_hcd and hf_hb_usb_2 debugging activated. Perhaps someone has a clue why this doesn't work.

[dmesg.log](https://github.com/jens-maus/RaspberryMatic/files/11455948/dmesg.log)

**Comment by jens-maus** [comment]
@reddevil156 thx for the log. In the endnof it one can see that the hb-rf-usb-2 is actually recognized and I guess that is also correctly appears via lsusb in proxmox, right? If so, what are the vm settings you are using for the usb? Did you try to disable/enable usb3 in the vm settings? And have you tried to debug within the runnibg RaspberryMatic if the usb-2 is showing uo there via lsusb correctly and if detect-radio-module is outputting anything sensible?!?

**Comment by reddevil156** [comment]
Thanks for your reply.

hb-rf-usb-2 is correctly detected in proxmox (via lsusb) and after passthrough also in raspberrymatic.
I used the vm settings from the installation script. I tried both ways with usb3.0 enabled and disabled. I did a test installation on another older computer (i5-4460) with a native usb 2.0 connector but the result is always the same.

The passthrough is working and the usb-2 is always correctly recognized. detect-radio-module is only sending the frames but there are no responses from the mod-pcb. When checking with the multimeter the mod-pcb doesn't have 3.3V, only 1.2V, so something in the configuration must be faulty.

**Comment by H3xF2x** [comment]
Hello all,

I tried a RPI-RF-MOD and had no problem at all:
`Mounting /data as /usr/local (Home Assistant Add-On): OK
Starting watchdog...
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Saving 256 bits of creditable seed for next boot
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.0, OK
Identifying Homematic RF-Hardware: ....HmRF: RPI-RF-MOD/HB-RF-USB-2@usb-fe980000.usb-1.1, HmIP: RPI-RF-MOD/HB-RF-USB-2@usb-fe980000.usb-1.1, OK
Updating Homematic RF-Hardware: RPI-RF-MOD: 4.2.14=>4.4.22, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .......OK
Starting rfd: .OK
Starting HMIPServer: ..............OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.69.7.20230506 (raspmatic_oci_arm64)`

Seems that the problem is specific for the HM-MOD-RPI-PCB/HM-RF-USB-2 and not the yellow.
